src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,395 @@
+     * @param command to perform
+     * @param verifyString string that should be present in a response payload
+     * @return boolean value indicating whether the operation is successful
+     */
+    private boolean executeAndVerify(String command, String verifyString) throws Exception {
+        String response = execute(command, verifyString);
+        return !StringUtils.isEmpty(response);
+    }
+
+    /**
+     * Execute the command and check the response to contain "verifyString" in a few attempts.
+     * It's not added into the general "CommandSuccess/Failure" list because the basic communicator
+     * checks for endsWith() but there's no guarantee that the response will end with a certain string,
+     * also sometimes there are multiple comands perfomed automatically on entry with multiple "** end" breaks.
+     * This part helps to have another layer of response verification.
+     *
+     * @param command to perform
+     * @param verifyString string that should be present in a response payload
+     * @return empty string or a command response, if successful
+     */
+    private String execute(String command, String verifyString) throws Exception {
+        String response;
+        controlOperationsLock.lock();
+        try {
+            updateLatestControlTimestamp();
+            refreshSshConnection();
+            response = send(command);
+            int retryAttempts = 0;
+            while (!response.contains(verifyString)){
+                retryAttempts++;
+                response = send(command);
+                if(retryAttempts >= 10) {
+                    return "";
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return response;
+    }
+
+    /**
+     * Check if the channel is connected and if not - reconnect
+     */
+    private void refreshSshConnection() throws Exception {
+        if (!isChannelConnected()) {
+            connect();
+        }
+    }
+
+    /**
+     * Update timestamp of the latest control operation
+     * */
+    private void updateLatestControlTimestamp(){
+        latestControlTimestamp = new Date().getTime();
+    }
+
+    /***
+     * Check whether the control operations cooldown has ended
+     * @return boolean value indicating whether the cooldown has ended or not
+     */
+    private boolean isValidControlCoolDown(){
+        return (new Date().getTime() - latestControlTimestamp) < controlsCooldownTimeout;
+    }
+
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        String meetingStatus = getCallStatus();
+        if(meetingStatus.equals("in meeting")){
+            logger.warn("Not able to connect. Meeting " + meetingNumber + " is in progress.");
+            return null;
+        } else if (meetingStatus.equals("connecting meeting")) {
+            callDisconnect();

```

Comment:
The problem is that sometimes the "connecting meeting" status may freeze indefinitely, if the meeting number is corrupted for instance. In this case ZR instance will basically freeze on this status (which will be also indicated with a spinner on ui saying "connecting"). So this is basically a recovery scenario here with an assumption that there are 2 possible scenarios this case may occur: **either** 2 similar connection requests are sent (like if the person dials an address and then does that right away one more time expecting immediate feedback), so in this case it will still connect, but the first request will be considered rather "stale", and even though the device is not stuck - the request will be still resolved successfully, **or** the initial request ended up having a typo or something, that led to an indefinite connection state, making the device not able to connect to anything else, unless disconnected. I'm not sure if it'd end up disconnecting itself after some time, I haven't tested it with long periods of time (only about 10 minutes, which is already quite a lot), but in order to make the device functional in such situations - it's disconnected. 
Sure, there's a third case - when there are just 2 different incoming dial requests going towards the same device within a short period of time (a couple of seconds which it would normally take) and then the first request will be disconnected due to a supposed conflict. Even though that such scenario is less likely to happen, still it is possible. 
So basically - this part here is a recovery process. Also, if there's no tablet-controller available atm - there's just no way to resolve this frozen connection status rather than using CLI (or alternatively we can add an extended properties control that would disconnect an app if it's "Connecting", but do not show the control when it's connected. Still not sure if this would make much sense to have a specific control for this case only). 

Screenshot below: the result of trying to join the meeting with id 1234567890. 
I have posted the question to Zoom staff in order to clarify if it's possible to retrieve the meeting number that instance is trying to connect to, though there's no such information provided in the documentation, call status is quite binary. 
![image](https://user-images.githubusercontent.com/59052678/97016421-61ac0900-1555-11eb-8bca-c9427a5fba92.png)

Reply:
If it can freeze indefinitely in connecting and that's the point, should we at least give it some time from the previous attempt to finish? It dos not appear to check anything, just disconnects. The good logic to me seems that if we are connecting to the same address as before, leave it along unless it is taking too long. And if it is too long - disconnect and try again.

Reply:
Agree, unfortunately we can't get the specific meetingId it's "connecting" to, so adding the workaround and javadocs to explain what's going on.